neptune/hierarchy_test.go

```diff
@@ -65,6 +67,103 @@ func TestNeptuneDB_GetCodesWithData(t *testing.T) {
+
+		poolMock := &internal.NeptunePoolMock{GetFunc: func(query string, bindings map[string]string, rebindings map[string]string) (vertices []graphson.Vertex, err error) {
+			return []graphson.Vertex{vertex}, nil
+		}}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			hierarchyExists, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
+
+			Convey("Then the expected query is sent to Neptune", func() {
+
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, testInstanceID, testDimensionName)
+				So(len(poolMock.GetCalls()), ShouldEqual, 1)
+				So(poolMock.GetCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the return value is true", func() {
+				So(hierarchyExists, ShouldBeTrue)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+
+	Convey("Given a mocked neptune DB that returns multiple hierarchy nodes", t, func() {
+
+		poolMock := &internal.NeptunePoolMock{GetFunc: func(query string, bindings map[string]string, rebindings map[string]string) (vertices []graphson.Vertex, err error) {
+			return []graphson.Vertex{vertex, vertex}, nil
+		}}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			_, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
+
+			Convey("Then the expected query is sent to Neptune", func() {
+
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, testInstanceID, testDimensionName)
+				So(len(poolMock.GetCalls()), ShouldEqual, 1)
+				So(poolMock.GetCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the expected error is returned", func() {
+				So(err, ShouldEqual, driver.ErrMultipleFound)
+			})
+		})
+	})
+
+	Convey("Given a mocked neptune DB that returns an empty array of vertices", t, func() {
+
+		poolMock := &internal.NeptunePoolMock{GetFunc: func(query string, bindings map[string]string, rebindings map[string]string) (vertices []graphson.Vertex, err error) {
+			return []graphson.Vertex{}, nil
+		}}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			hierarchyExists, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
+
+			Convey("Then the expected query is sent to Neptune", func() {
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, testInstanceID, testDimensionName)
+				So(len(poolMock.GetCalls()), ShouldEqual, 1)
+				So(poolMock.GetCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the return value is false", func() {
+				So(hierarchyExists, ShouldBeFalse)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+
+	Convey("Given a mocked neptune DB that returns an error", t, func() {
+
+		poolMock := &internal.NeptunePoolMock{
+			GetFunc: internal.ReturnMalformedNilInterfaceRequestErr,
+		}
+		db := mockDB(poolMock)
+
+		Convey("When HierarchyExists is called", func() {
+			_, err := db.HierarchyExists(ctx, testInstanceID, testDimensionName)
```

Comment:
Same here: check the bool return value as it is an output of the function being tested.